onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_info.cc

```diff
@@ -360,6 +360,9 @@ void TensorrtExecutionProviderInfo::UpdateProviderOptions(void* provider_options
   trt_provider_options_v2.trt_engine_hw_compatible = internal_options.engine_hw_compatible;
   trt_provider_options_v2.trt_onnx_bytestream = internal_options.onnx_bytestream;
   trt_provider_options_v2.trt_onnx_bytestream_size = internal_options.onnx_bytestream_size;
-  trt_provider_options_v2.trt_op_types_to_exclude = copy_string_if_needed(internal_options.op_types_to_exclude);
+  if (options.find("trt_op_types_to_exclude") != options.end()) {
+    trt_provider_options_v2.trt_op_types_to_exclude = copy_string_if_needed(internal_options.op_types_to_exclude);
+    trt_provider_options_v2.trt_op_types_to_exclude_str_is_dynamic_allocation = 1;
+  }
```

Comment:
what if we dynamically allocate the ""NonMaxSuppression,NonZero,RoiAlign" string in the else case here if we don't find trt_op_types_to_exclude in options. then we would always deallocate it.

Reply:
> what if we dynamically allocate the ""NonMaxSuppression,NonZero,RoiAlign" string in the else case here if we don't find trt_op_types_to_exclude in options.

If user uses OrtApis::UpdateTensorRTProviderOptions to update the OrtTensorRTProviderOptionsV2 instance, i don't think the case you mention exist. i assume you mean the default value "NonMaxSuppression,NonZero,RoiAlign" that TRT EP allocates, and it's static allocation, no need to deallocate. it will automatically free by the program.